docs/reference/mapping/types.asciidoc

```diff
@@ -1,72 +1,208 @@
 [[mapping-types]]
-== Field data types
+== Field data types [[_core_datatypes]]
+
+Each field has a _field data type_, or _field type_. This type indicates the
+kind of data the field contains, such as strings or boolean values, and its
+intended use. For example, you can index strings to both `text` and `keyword`
+fields. However, `text` field values are <<analysis,analyzed>> for full-text
+search while `keyword` strings are left as-is for filtering and sorting.
+
+Field types are grouped by _family_. Types in the same family behave similarly
+and can be used interchangeably in your application. For example, the `keyword`
+family consists of the `keyword`, `constant_keyword`, and `wildcard` field
+types. In most cases, the type family is the same as the field type. For
+example, the `boolean` type family consists of one field type: `boolean`.
+
+You can retrieve a field's type family using the <<search-field-caps,field
+capabilities API>>.
+
+The following sections organize the available field types based on their
+intended use. Except where noted, the type family is the same as the field type.
+
+<<binary,`binary`>>::
```

Comment:
For me this organization is a bit unclear. Some types like `binary` and `boolean` don't belong to an overall section. And the sections don't relate to a consistent concept -- some describe use cases like 'full text search' while others are broad type categories like 'dates'.

It's actually quite challenging to come up with a good grouping! I'll reach out to you through another channel with a brainstorming document.

Reply:
Awesome! Thanks!